eth/gasprice/locaFeeMarket/nativediscount.go

```diff
@@ -0,0 +1,318 @@
+package locaFeeMarket
+
+import (
+	"bytes"
+	"errors"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	"math/big"
+)
+
+
+const (
+	historicalMeanGasPrice = 35783571428 //wei
+	historicalSDGasPrice = 849870638 //wei
+	maxDiscountedGasPrice  = historicalMeanGasPrice + historicalSDGasPrice/2 // =36208506747wei
+
+	coreTransferGasAmount = 21000 // hard-coded gas amount
+
+	functionSelectorSize = 4
+	transferDataLen = 4+32+32  // transfer(address,uint256): 4 bytes for func ID, 32 bytes for address (20 bytes padded to 32), 32 bytes for amount
+	transferFromDataLen = 4+32+32+32  // transferFrom(address,address,uint256): 4 bytes for func ID, 32 bytes for from address, 32 bytes for to address, 32 bytes for amount
+	approveDataLen = 4+32+32// approve(address,uint256): 4 bytes for func ID, 32 bytes for the spender address (padded), 32 bytes for the amount
+
+	supportWhitelistedErc20s = true
+	useDebugDiscountedGasPrice = false
+)
+
+// erc20 function types eligible for discount
+type funcType uint
+
+const (
+	ftype_none funcType = iota
+	ftype_transfer
+	ftype_transferFrom
+	ftype_approve
+)
+
+var (
+	// erc20's transfer: the first 4 bytes of the keccak256 hash of "transfer(address,uint256)"
+	transferFunction = []byte{0xa9, 0x05, 0x9c, 0xbb}
+
+	// erc20's transferFrom: the first 4 bytes of the keccak256 hash of "transferFrom(address,address,uint256)"
+	transferFromFunction = []byte{0x23, 0xb8, 0x72, 0xdd}
+
+	// erc20's approve: the first 4 bytes of the keccak256 hash of "approve(address,uint256)"
+	approveFunction = []byte{0x09, 0x5e, 0xa7, 0xb3}
+
+	debugDiscountedGasPrice = new(big.Int).SetUint64(18_000_000_000)
+
+	errNotFunctionInvocation = errors.New("tx is not a function invocation")
+
+	//-------
+	btcLstContract = common.HexToAddress("0xTBD")
+
+	internalErc20Whitelist = map[common.Address]bool{
+		btcLstContract: true,
+	}
+)
+
+
+func AdjustGasPriceForEstimation(_networkGasPrice *hexutil.Big, _gasAmount *hexutil.Uint64, _value *hexutil.Big, to *common.Address, data []byte) *hexutil.Big {
```

Comment:
Why are some function arguments prefixed with `_` here?

Reply:
to allow using non-underscored names inside function